internal/gatewayapi/listener.go

```diff
@@ -288,3 +314,45 @@ func processMetrics(envoyproxy *egv1a1.EnvoyProxy) *ir.Metrics {
 		EnableVirtualHostStats: envoyproxy.Spec.Telemetry.Metrics.EnableVirtualHostStats,
 	}
 }
+
+func (t *Translator) processServiceDestination(backendRef egv1a1.BackendRef, protocol ir.AppProtocol, envoyproxy *egv1a1.EnvoyProxy, resources *Resources) *ir.DestinationSetting {
+	var (
+		endpoints   []*ir.DestinationEndpoint
+		addrType    *ir.DestinationAddressType
+		servicePort v1.ServicePort
+		backendTLS  *ir.TLSUpstreamConfig
+	)
+
+	// TODO (davidalger) Handle case where Service referenced by backendRef doesn't exist
+	serviceNamespace := NamespaceDerefOr(backendRef.Namespace, envoyproxy.Namespace)
+	service := resources.GetService(serviceNamespace, string(backendRef.Name))
+	for _, port := range service.Spec.Ports {
```

Comment:
This will panic if the backendRef refers to a Service which cannot be found. Ideally this would surface as a message on the status somewhere, similar to how it does on XRoutes.

Status isn't currently being used on EnvoyProxy as far as I can tell. It seems it could make sense to put errors such as this on the Gateway rather than the EnvoyProxy since it's something which will arise when processing the Gateway.

@arkodg @zirain Do either of you have thoughts on where errors such as this should be surfaced?

Reply:
ah
1. any EnvoyProxy errors are surfaced in the `GatewayClass` , lets tackle this separately, and raise a issue to track this because moving it to `Accepted=False` will break everything in the next reconciliation :) 
2. For now lets stick a Direct Response that returns `500` similar to https://github.com/envoyproxy/gateway/blob/2c4d385fcfb02841483cc36e0292175f60249bbb/internal/gatewayapi/route.go#L492